dspace-api/src/main/java/org/dspace/core/HibernateDBConnection.java

```diff
@@ -203,44 +206,38 @@ private void configureDatabaseMode() throws SQLException {
                         uncacheEntity(bitstream);
                     }
                 }
-            //} else if(entity instanceof Bitstream) {
-            //    Bitstream bitstream = (Bitstream) entity;
-            //    No specific child entities to decache
+
+            // BITSTREAM
+            // No specific child entities to decache
+
+            // COMMUNITY
             } else if (entity instanceof Community) {
                 Community community = (Community) entity;
-                if(Hibernate.isInitialized(community.getAdministrators())) {
-                    uncacheEntity(community.getAdministrators());
-                }
+
+                //We don't uncache groups as they might still be referenced from the Context object
 
                 if(Hibernate.isInitialized(community.getLogo())) {
                     uncacheEntity(community.getLogo());
                 }
+
+            // COLLECTION
             } else if (entity instanceof Collection) {
                 Collection collection = (Collection) entity;
+
+                //We don't uncache groups as they might still be referenced from the Context object
+
                 if(Hibernate.isInitialized(collection.getLogo())) {
                     uncacheEntity(collection.getLogo());
                 }
-                if(Hibernate.isInitialized(collection.getAdministrators())) {
-                    uncacheEntity(collection.getAdministrators());
-                }
-                if(Hibernate.isInitialized(collection.getSubmitters())) {
-                    uncacheEntity(collection.getSubmitters());
-                }
                 if(Hibernate.isInitialized(collection.getTemplateItem())) {
                     uncacheEntity(collection.getTemplateItem());
                 }
-                if(Hibernate.isInitialized(collection.getWorkflowStep1())) {
-                    uncacheEntity(collection.getWorkflowStep1());
-                }
-                if(Hibernate.isInitialized(collection.getWorkflowStep2())) {
-                    uncacheEntity(collection.getWorkflowStep2());
-                }
-                if(Hibernate.isInitialized(collection.getWorkflowStep3())) {
-                    uncacheEntity(collection.getWorkflowStep3());
-                }
             }
 
+            if(!readOnlyEnabled && getSession().isDirty()) {
+                getSession().flush();
+            }
```

Comment:
Here's how to recreate the `StaleStateException` mentioned above:
1. Locally apply this PR to the latest dspace-6_x branch (rebuild/redeploy)
2. Enable the 'authority' consumer in your dspace.cfg
3. Submit a new item. After completing the submission, you'll see the `StaleStateException` thrown (via the AuthorityConsumer).  Also, the new item won't be indexed (as discovery consumer seems to not be triggered after the exception occurs).

However, it's worth noting that with the "authority" consumer disabled (which is default), everything seems to work properly.